### ISSUE TITLE
`nuxt generate` hangs when page has a recursive function that auto executes until that function completes

### ISSUE THREAD

**Issue by Chalks** [nuxt/nuxt#29176]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.10.0`
- Nuxt Version:     `3.13.2`
- CLI Version:      `3.14.0`
- Nitro Version:    `2.9.7`
- Package Manager:  `npm@10.2.3`
- Builder:          `-`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-id9dsp?file=pages%2Ftest.vue

When you `npm run build` or `npm run dev` the page `/test` works fine. If however you `npm run generate` it hangs at this point for the duration of the recursion:

![Screenshot 2024-09-26 at 9 50 01 AM](https://github.com/user-attachments/assets/0f33cdca-5ae5-4049-920f-cc676f5b06a8)

### Describe the bug

If you have a recursive function that utilizes `setTimeout` to do some action on page load (like start a countdown) then generate hangs until the recursive function has completely finished. A think there are a few issues with this:

1. the output looks like it completed successfully, but it hasn't, and there's no way to understand what it's waiting for without debugging (adding `debug: true` doesn't change the output meaningfully in this case). I don't think it should show that `/test` successfully prerendered in 44ms when I actually have to wait 100 seconds (given the code in the example) for it to finish.
2. should setTimeouts affect rendering?
3. I don't see any see anything that might tip me off about this in [the documentation](https://vuejs.org/api/sfc-script-setup) about `<script setup>`

**Comment by BobbieGoede** [comment]
It seems like it's not hanging while rendering, but waiting for the `setTimeout` events to finish server-side. 

I doubt this is intended behavior but I don't know if it is unexpected or a known quirk 😅 

In this specific use case you likely only want this countdown to happen client-side by moving the `countdown()` invocation inside `onMounted`:

```ts
onMounted(() => {
  if (auto) {
    countdown();
  }
});
```

This should avoid the server from hanging

**Comment by luc122c** [comment]
@danielroe I don't think this is a bug in Nuxt or Nitro. As @BobbieGoede says, the renderer is waiting for the page to settle but it never does. I think the workaround is actually the best-practice way to handle this. Perhaps It might be worth adding a note to the docs to point out this caveat and workaround?

**Comment by danielroe** [comment]
agreed. I think we can close.

if you think there's an issue here, this is nitro behaviour and you could raise an issue there.

nuxtathon closed @luc122c
